src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +192,55 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.push(
+      `?hazardLayerId=${hazardLayerId}&baselineLayerId=${baselineLayerId}&date=${selectedDate}&statistic=${statistic}&aboveThreshold=${aboveThreshold}&belowThreshold=${belowThreshold}`,
+    );
   };
 
+  useEffect(() => {
+    const queryString = history.location.search;
+    const params = new URLSearchParams(queryString);
+    const hazardLayerParamId = params.get('hazardLayerId');
+    const baselineLayerParamId = params.get('baselineLayerId');
+    const selectedParamDate: number = Number(params.get('date'));
+    const statisticParam = params.get('statistic');
+    const aboveThresholdParam: string = params.get('aboveThreshold') || '';
+    const belowThresholdParam: string = params.get('belowThreshold') || '';
+
+    if (!hazardLayerParamId || !baselineLayerParamId || !statisticParam) {
+      return;
+    }
+    function isHazardLayerId(layer: string): layer is LayerKey {
+      return typeof layer === 'string';
+    }
+    function isBaselineLAyerId(layer: string): layer is LayerKey {
```

Comment:
```suggestion
    function isBaselineLayerId(layer: string): layer is LayerKey {
```

Reply:
If you are stuck on how to complete these functions, I recommend checking the key exists here: https://github.com/WFP-VAM/prism-frontend/blob/master/src/config/utils.ts#L116, then checking the `layer.type` to verify its the correct type (Hazard - WMS or Baseline - NSO)

Reply:
> If you are stuck on how to complete these functions, I recommend checking the key exists here: https://github.com/WFP-VAM/prism-frontend/blob/master/src/config/utils.ts#L116, then checking the `layer.type` to verify its the correct type (Hazard - WMS or Baseline - NSO)

The function is already complete, It's doing what I wanted that is to check whether the passed argument is a correct Layer ( `"ndvi" | "ModisLST" | "ModisVHI" ... `)

Reply:
Are you sure checking for a string type is good enough? We could pass absolutely anything like `not_a_layer` and cause a nasty error somewhere down the line. I tried just now, and it crashed the application without a graceful error/warning.

I recommend checking if the string is actually valid using the link I sent so we can verify the layer actually exists, and follow with a non-action or a graceful error.